### PR TITLE
Add a days_in_month accessor to CFTimeIndex

### PR DESCRIPTION
**Pull request by spencerkclark**
- [x] Tests added
 - [x] Passes `isort -rc . && black . && mypy . && flake8`
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

This adds a `days_in_month` accessor to CFTimeIndex, which allows for easy computation of monthly time weights for non-standard calendars:
```
In [1]: import xarray as xr

In [2]: times = xr.cftime_range("2000", periods=24, freq="MS", calendar="noleap")

In [3]: da = xr.DataArray(times, dims=["time"])

In [4]: da.dt.days_in_month
Out[4]:
<xarray.DataArray 'days_in_month' (time: 24)>
array([31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31, 31, 28, 31, 30, 31,
       30, 31, 31, 30, 31, 30, 31])
Coordinates:
  * time     (time) object 2000-01-01 00:00:00 ... 2001-12-01 00:00:00
```

This simplifies the ["Calculating Seasonal Averages from Timeseries of Monthly Means" example](http://xarray.pydata.org/en/stable/examples/monthly-means.html) @jhamman wrote for the docs a while back, which I've taken the liberty of updating.

The ability to add this feature to xarray is thanks in large part to @huard, who added a `daysinmonth` attribute to `cftime.datetime` objects late last year: https://github.com/Unidata/cftime/pull/138.

### CODE REVIEW

File: doc/whats-new.rst [hunk 1]
```diff
@@ -48,6 +48,13 @@ New Features
   By `Todd Jennings <https://github.com/toddrjen>`_
 - Allow plotting of boolean arrays. (:pull:`3766`)
   By `Marek Jacob <https://github.com/MeraX>`_
+- A ``days_in_month`` accessor for :py:class:`xarray.CFTimeIndex`, analogous to
+  the ``days_in_month`` accessor for a :py:class:`pandas.DatetimeIndex`, which
+  returns the days in the month each datetime in the index.  Now days in month
+  weights for both standard and non-standard calendars can be obtained using
+  the :py:class:`~core.accessor_dt.DatetimeAccessor` (:pull:`3935`).  This
+  feature requires cftime version 1.1.0 or greater.  By
+  `Spencer Clark <https://github.com/spencerkclark>`_.
```
Comment (dcherian) [inline doc/whats-new.rst:57]: Let's also mention the updated example notebook under "Documentation"?

**Comment by spencerkclark** [comment]
> Can you strip outputs from the notebook? That should reduce the diff here and later when it gets updated next.

That didn't occur to me, thanks.  That's indeed much better.  I added a documentation note as well; things should hopefully be all set now.

**Comment by jhamman** [comment]
Fantastic! Thanks @spencerkclark!

File: doc/examples/monthly-means.ipynb [hunk 2]
```diff
@@ -183,7 +108,7 @@
    },
    "outputs": [],
    "source": [
-    "print(ds_weighted)"
+    "ds_weighted"
```
Comment (jhamman) [inline doc/examples/monthly-means.ipynb:111]: would this cell benefit from an update using our new weighted api?

Comment (spencerkclark) [reply doc/examples/monthly-means.ipynb:111]: Thanks @jhamman!  I considered that, but wasn't quite sure how to work it in with the `groupby` API.  Happy to take suggestions, however.

Comment (mathause) [reply doc/examples/monthly-means.ipynb:111]: Neither do I to be honest... #3937

Comment (dcherian) [reply doc/examples/monthly-means.ipynb:111]: looks like this will need some engineering so things look good to me for now.

**Comment by dcherian** [comment]
Thanks @spencerkclark

**Pull request merged**